salt/states/boto_iam.py

```diff
@@ -0,0 +1,417 @@
+# -*- coding: utf-8 -*-
+'''
+Manage IAM roles.
+=================
+
+.. versionadded:: TBD
+
+This module uses ``boto``, which can be installed via package, or pip.
+
+This module accepts explicit IAM credentials but can also utilize
+IAM roles assigned to the instance through Instance Profiles. Dynamic
+credentials are then automatically obtained from AWS API and no further
+configuration is necessary. More information available `here
+<http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
+
+It's also possible to specify ``key``, ``keyid`` and ``region`` via a profile, either
+passed in as a dict, or as a string to pull from pillars or minion config:
+
+.. code-block:: yaml
+
+    create-user:
+      boto_iam.user_present:
+        - name: myuser
+        - password: NewPassword$$1
+        - group: mygroup
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'fdkjsafkljsASSADFalkfjasdf'
+
+.. code-block:: yaml
+
+    create-group:
+      boto_iam.group_present:
+        - name: mygroup
+        - users:
+          - myuser
+          - myuser1
+        - policy_name: test1
+        - policy: '{ "Version": "2012-10-17", "Statement": [ { "Effect": "Allow", "Action": "*", "Resource": "*" }]}'
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'safsdfsal;fdkjsafkljsASSADFalkfj'
+
+.. code-block:: yaml
+
+    change-policy:
+      boto_iam.account_policy:
+        - change_password: True
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'safsdfsal;fdkjsafkljsASSADFalkfj'
+
+.. code-block:: yaml
+
+    create server certificate:
+      boto_iam.server_cert_present:
+        - name: mycert
+        - public_key: salt://base/mycert.crt
+        - private_key: salt://base/mycert.key
+        - cert_chain: salt://base/mycert_chain.crt
+        - region: eu-west-1
+        - keyid: 'AKIAJHTMIQ2ASDFLASDF'
+        - key: 'fdkjsafkljsASSADFalkfjasdf'
+
+.. code-block:: yaml
+    delete server certificate:
+      boto_iam.server_cert_absent:
+        - name: mycert
+'''
+
+import logging
+import json
+import salt.utils.odict as odict
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if boto is available.
+    '''
+    return 'boto_iam' if 'boto_iam.get_user' in __salt__ else False
```

Comment:
I am not sure how this virtual actually works, but so far I saw that it either returns a name of the module that has to be accessed or False, if module is not available. Does True means that the name will be identical to the name of the file of the module? If yes, then you  are right.